solidity/contracts/RepCrowdsourcer.sol

```diff
@@ -7,49 +7,42 @@ interface IERC20 {
     function transferFrom(address from, address to, uint256 value) external returns (bool);
 }
 
-contract RepCrowdsourcer {
+contract GoFundMicah {
 	IERC20 public repV2 = IERC20(0x221657776846890989a759BA2973e427DfF5C9bB);
 	address public micahAddress = 0x82c34Fdbc5c71B899F484e716BAD2271e2c6f0C3; // micah.darkflorist.eth VERIFY!
 	uint256 public minBalanceToWithdraw = 200000 ether;
 
 	mapping(address => uint256) public deposits;
-	bool public contractClosed;
+	bool public contractClosed = false;
 
 	event Deposit(address indexed depositor, uint256 amount);
 	event Withdraw(address indexed withdrawer, uint256 amount);
 	event ContractClosed();
 	event MicahWithdraw(uint256 amount);
 
-	bool internal locked;
-
-	modifier noReentrant() {
-		require(!locked, 'No re-entrancy');
-		locked = true;
-		_;
-		locked = false;
-	}
-
-	function deposit(uint256 amount) public noReentrant {
+	function deposit(uint256 amount) public {
 		require(!contractClosed, 'Deposits are closed');
 		repV2.transferFrom(msg.sender, address(this), amount);
 		deposits[msg.sender] += amount;
 		emit Deposit(msg.sender, amount);
 	}
 
-	function withdraw(uint256 amount) public noReentrant {
-		require(deposits[msg.sender] >= amount, 'not enough balance to withdraw');
-		deposits[msg.sender] -= amount;
-		repV2.transfer(msg.sender, amount);
-		emit Withdraw(msg.sender, amount);
+	function internalWithdraw(address recipient) private {
+		uint256 amount = deposits[recipient];
+		require(amount > 0, 'not enough balance to withdraw');
```

Comment:
Is this line actually necessary?  REP may already error if you try to withdraw 0, and even if it doesn't is there a problem with a 0 amount withdraw?  Costs a little bit more gas, but in the `massWithdraw` scenario it would be nice to not fail the entire batch.
```suggestion
```

Reply:
it sends 0 rep. I wondered to block that, but its not mandatory